README.rst

```diff
@@ -1,6 +1,130 @@
-==========================
- REANA Demo - Hello World
-==========================
+====================================
+ REANA demo example - "hello world"
+====================================
 
-This repository provides a simple "hello world" example demonstrating data
-analysis encapsulation for running it via the REANA system.
+About
+=====
+
+This repository provides a simple "hello world" application example for `REANA
+<http://reanahub.io/>`_ reusable research data analysis plaftorm.
+
+Code
+====
+
+A simple "hello world" Python code for illustration:
+
+- `helloworld.py <helloworld.py>`_
+
+The program takes several optional arguments such as the name of the person to
+greet and an optional sleeptime period and produces an output file with the
+greeting:
+
+.. code-block:: console
+
+   $ python helloworld.py --name Jane --sleeptime 0
+   $ cat output/greetings.txt
+   Hello Jane!
+
+Containerisation
+================
+
+In order to be able to rerun the code even several years in the future, we need
+to "encapsulate the current environment" for example to freeze the Python
+version the code is using. We shall achieve this by preparing a `Docker
+<https://www.docker.com/>`_ container image for the program:
+
+- `Dockerfile <Dockerfile>`_
+
+For example:
+
+.. code-block:: console
+
+    $ cat Dockerfile
+    FROM python:2.7-alpine
+    ADD . /code
+    WORKDIR /code
+    ENTRYPOINT ["python", "helloworld.py"]
+    CMD ["--name", "John Doe", \
+         "--outputfile", "output/greetings.txt", \
+         "--sleeptime", "3"]
+
+We build the container image:
```

Comment:
_We build the container image and tag it as 'helloworld' -image (Note the dot at end of the command):_